### DISCUSSION TITLE
Using prisma migate to update prod db schema

### DISCUSSION THREAD

**Discussion by dinakart** [vercel/community#4023] [Help]
### Summary

Hey everyone! I am looking for an easy way to edit my postgres production schema with prisma WITHOUT any data loss. I tried to run **yarn prisma migrate dev** with the postgres db url but am running into an issue

**_Error: db error: ERROR: database "[DB NAME]" is being accessed by other users
DETAIL: There are 2 other sessions using the database.
   0: migration_core::state::DevDiagnostic
             at migration-engine/core/src/state.rs:269_**

However, I am not able to kill any sessions in psql because my account is not a superuser and I am not sure how to log into the superuser to perform these actions (since my role is not one).

Is this the best way to edit prod table schema through prisma or is there a better way of doing this? I basically want a way for my local db to be in sync with my vercel postgres production db. I also tried yarn **prisma db push** but that deleted all my production data, luckily I saved everything locally first. 

Thank you in advance!

### Steps to Reproduce

N/A

**Comment by kleenkanteen** [comment]
Your schema.prisma should be defined like this:

```
datasource db {
  provider = "postgresql"
  url = env("DATABASE_URL") // uses connection pooling
  directUrl = env("POSTGRES_URL_NON_POOLING") // uses a direct connection
}
```
It needs the `url` as well as the `directUrl`. When I left out the latter, I got the same error as you. If you use vercel postgres both are given.

**Reply by Susanta-0007** [reply]
![Screenshot 2024-01-28 225232](https://github.com/vercel/community/assets/89796600/cd8f3fd0-600a-4948-b4ab-bf53ff4af1fc)
Same error here .  how to fix follow the up steps but show same error

**Reply by kleenkanteen** [reply]
@Susanta-0007 Did you follow what I suggested?

**Reply by Susanta-0007** [reply]
Yes

**Comment by kleenkanteen** [comment]
Where is your database hosted and try asking phind

**Reply by Susanta-0007** [reply]
Superbase

**Comment by kleenkanteen** [comment]
Make a new db
